mlir/lib/Target/LLVMIR/DebugImporter.cpp

```diff
@@ -309,6 +309,35 @@ DICommonBlockAttr DebugImporter::translateImpl(llvm::DICommonBlock *node) {
                                 translate(node->getFile()), node->getLineNo());
 }
 
+DIGenericSubrangeAttr
+DebugImporter::translateImpl(llvm::DIGenericSubrange *node) {
+  auto getAttrOrNull =
+      [&](llvm::DIGenericSubrange::BoundType data) -> Attribute {
+    if (data.isNull())
+      return nullptr;
+    if (auto *expr = dyn_cast<llvm::DIExpression *>(data))
+      return translateExpression(expr);
+    if (auto *var = dyn_cast<llvm::DIVariable *>(data)) {
+      if (auto *local = dyn_cast<llvm::DILocalVariable>(var))
+        return translate(local);
+      if (auto *global = dyn_cast<llvm::DIGlobalVariable>(var))
+        return translate(global);
+      return nullptr;
+    }
+    return nullptr;
+  };
+  Attribute count = getAttrOrNull(node->getCount());
+  Attribute upperBound = getAttrOrNull(node->getUpperBound());
+  Attribute lowerBound = getAttrOrNull(node->getLowerBound());
+  Attribute stride = getAttrOrNull(node->getStride());
+  // DIGenericSubrangeAttr requires lowerBound, stride and one of count or
+  // upperBound.
+  if (!lowerBound || !stride || (!count && !upperBound))
+    return {};
```

Comment:
```suggestion
  // Either count or the upper bound needs to be present. Otherwise, the
  // metadata is invalid.
  if (!count && !upperBound)
    return {};
```
Should we just check either count or upper bound is set here as above? I assume the attribute verifier checks the other two are non-null?

Reply:
In the llvm verifier, there is a slight difference between `DISubrange` and `DIGenericSubrange`. The `lowerBound` and `stride` fields can be null for `DISubrange` but not for `DIGenericSubrange`. Please see [Verifier::visitDIGenericSubrange](https://github.com/llvm/llvm-project/blob/ca1154d1d41c75db6594428a8cdf263cf7041896/llvm/lib/IR/Verifier.cpp#L1184). I was trying to replicate the same check here.

Reply:
I would expect that the attribute verifier of DIGenericSubrangeAttr already verifies that lowerBound and stride are non-null since they are not optional attribute parameters? There have been issues with attribute verifiers in the past, so I am not a 100% sure this works as expected (if you roundtrip an attribute without a stride or lowerBound does this produce an error?). Anyways, if you prefer the more complete logic I am fine with it. 

Note that long term verifiers are preferable over silently dropping stuff during the import. Especially since verifiers also work when generating the debug info in MLIR. There may still be technical issues with attribute verifiers though.